### ISSUE TITLE
Zip installation through Magisk Manager partially wiped device (rare bug?)

### ISSUE THREAD

**Issue by Didgeridoohan** [topjohnwu/Magisk#46]
**Edit 2017-01-15:**
Added additional/updated information in comment below.

**Original text:**
I installed metaspooks ViPER-FX module v4.1 through the Magisk Manager. After the reboot, root was lost and parts of my device had been wiped clean.

Googles Messenger (sms) had been almost completely reset with all text messages deleted and notification sounds lost.

In Google calendar, the connection to outside calendars had been reset and needed to be resynced.

All app permissions had also been reset.

I'm providing the Magisk log prior to the wipe, where I had uninstalled v4.0 of the ViPER-FX module (it had a faulty id that made it look like topjohnwu's v4a module). There's also the Magisk log from installing v4.1. Unfortunately I have no more logs available, but I'm also providing the module in question. If needed I could run the installation again and try to catch some logs of the process, but resetting my phone after the wipe was a real pain...

[wipe.zip](https://github.com/topjohnwu/Magisk/files/705941/wipe.zip)
[viperfx-mkm-v4.1-magisk.zip](https://github.com/topjohnwu/Magisk/files/705945/viperfx-mkm-v4.1-magisk.zip)

I'm also attaching logs from installing the module in TWRP (also leading to root being lost).

[twrp.zip](https://github.com/topjohnwu/Magisk/files/705942/twrp.zip)

Device is a Nexus 6 with stock 7.1.1 (january patch), Franco Kernel r60 and Magisk v10.2 with phh's superuser topjohnwu r2.

I did see a similar thing (sms deleted, permissions reset) happening on an earlier release of Magisk (maybe v6), but I did not collect any data at that time, unfortunately.

**Comment by Didgeridoohan** [comment]
Update on the earlier incident I mentioned. I just remembered how it happened that time.

I believe it was on Magisk v6 and as a result of flashing the uninstaller script my device got partially wiped in the exact same manner as described above.

**Comment by Didgeridoohan** [comment]
Ok... Scratch everything... In an attempt to actually catch some useful logs of the incident, it turns out to not be reproducible. Flashing the same module that failed miserably when installing through Magisk Manager and caused root to disappear when flashed through TWRP, suddenly worked like a charm.

Installed it through the Magisk Manager again and after the reboot everything worked just fine.

24-48 hours ago flash failure through TWRP was completely reproducible. Today, not so much. 

Something must have happened when Magisk merged the new module and messed up my system. The wipe happened and it also turned out that my Google account completely stopped syncing, requiring a reset. I was also unable to flash a module replacing/adding files to /system (ui audio and notifications), that had previously worked without a hitch. Everytime root would be lost on reboot.

Today, everything suddenly works beautifully again and I don't know why.

Since the incident with partial device wipe I have several times uninstalled and installed Magisk trying to hunt down the problem in issue #45 and I had sort of given up on getting my audio file replacements to work for the time being.

After flashing the file module again in Magisk Manager and seeing that everything was working I also tried flashing my audio file replacement module. Lo and behold when that also turned out just fine. Feeling brave I also decided to flash the problematic kernel described in issue #45 and even that worked like a charm.

Suddenly none of the problems I have been battling with for the two past days were an issue anymore.

There's one difference that might have something to do with the problem and that's the zip file I flashed. The first time I had downloaded it on my phone and the second time on my computer and then transferred it to my device. Maybe the first download had gotten corrupted or damaged somehow and this caused a bad Magisk merge resulting in this chain of events (pure speculation I have absolutely no idea)? And could then a proper Magisk merge have reset everything?

I'm gonna close issue #45, since it's no longer an issue, but I'll leave the reference here since there seems to be something in Magisk that can cause this partial wipe in certain circumstances.
